test/integration/hook-err.spec.js

```diff
@@ -267,6 +267,52 @@ describe("hook error handling", function () {
     });
   });
 
```

Comment:
[Testing] It's possible for tests to throw something that's not an `Error`. It's not common, but within that the most common culprit's I've seen are either `null` or a `string`. Test error handling code tends to be brittle about this annoying edge case. Could you please add a test for each of those two types of non-`Error` exceptions?